### PR TITLE
Added `no_data_value_u64`, `set_no_data_value_u64`, `no_data_value_i64` and `set_no_data_value_i64` to `RasterBand`

### PR DESCRIPTION
**Pull request by metasim**
- [X] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [X] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

Adds `RasterBand::no_data_value_u64` and `RasterBand::no_data_value_i64`, etc.

### CODE REVIEW

**Comment by lnicola** [comment]
Can you resample those to 1x1 and maybe merge them into a single file?

File: src/raster/rasterband.rs [hunk 1]
```diff
@@ -696,7 +696,61 @@ impl<'a> RasterBand<'a> {
         None
     }
 
-    /// Set the no data value of this band.
+    /// Fetch the no-data value for this band.
+    ///
+    /// This method should ONLY be called on rasters whose data type is `UInt64`.
+    ///
+    /// If there is no no-data value, an out of range value will generally be returned.
```
Comment (lnicola) [inline src/raster/rasterband.rs:703]: I don't think this makes sense.

File: src/raster/rasterband.rs [hunk 2]
```diff
@@ -696,7 +696,61 @@ impl<'a> RasterBand<'a> {
...
+            return Some(no_data);
+        }
+        None
+    }
+
+    /// Fetch the no data value for this band.
+    ///
+    /// This method should ONLY be called on rasters whose data type is `Int64`.
+    ///
+    /// If there is no out of data value, an out of range value will generally be returned.
```
Comment (lnicola) [inline src/raster/rasterband.rs:730]: Same, but worse.

Comment (metasim) [reply src/raster/rasterband.rs:730]: It's what the docs say 🤷

Comment (metasim) [reply src/raster/rasterband.rs:730]: Should I ask `gdal-dev`? Or just strike it?

**Comment by metasim** [comment]
Just realized I should add the setters too. Not enough coffee this morning...

File: src/raster/rasterband.rs (hunk 2 above)
Comment (lnicola) [reply src/raster/rasterband.rs:730]: We return `None`, right?

**Comment by lnicola** [comment]
Good idea, I was actually wondering if there's any way to set it. Might make sense to use a larger value then, like 2^62.

**Comment by metasim** [comment]
> Can you resample those to 1x1 and maybe merge them into a single file?

How about a 1x2 so we can have a data value and a no-data value?

File: CHANGES.md [hunk 3]
```diff
@@ -1,6 +1,9 @@
 # Changes
 
 ## Unreleased
+- Added bindings for `GetNoDataValueAsInt64` and `GetNoDataValueAsUInt64`
```
Comment (lnicola) [inline CHANGES.md:4]: Let's spell out the method names here.

**Comment by metasim** [comment]
> Good idea, I was actually wondering if there's any way to set it. Might make sense to use a larger value then, like 2^62.

I basically neglected to see these

* https://gdal.org/api/gdalrasterband_cpp.html#_CPPv4N14GDALRasterBand21SetNoDataValueAsInt64E7int64_t
* https://gdal.org/api/gdalrasterband_cpp.html#_CPPv4N14GDALRasterBand22SetNoDataValueAsUInt64E8uint64_t
*

**Comment by metasim** [comment]
> Doesn't seem necessary to me...

What are you referring to here? The comment sequence is confusing.

**Comment by lnicola** [comment]
I meant that about the second pixel with a valid value. I don't think it's needed to test these functions, but they could be useful in a simple warping/resampling test.

File: src/raster/rasterband.rs [hunk 4]
```diff
@@ -713,6 +713,101 @@ impl<'a> RasterBand<'a> {
...
+    ///
+    /// If `no_data` is `None`, any existing no-data value is deleted.
+    ///
+    /// # Notes
+    /// See also:
+    /// [`GDALSetRasterNoDataValueAsInt64`](https://gdal.org/api/raster_c_api.html#_CPPv431GDALSetRasterNoDataValueAsInt6415GDALRasterBandH7int64_t),
+    /// [`GDALDeleteRasterNoDataValue`](https://gdal.org/api/raster_c_api.html#_CPPv427GDALDeleteRasterNoDataValue15GDALRasterBandH)
+    #[cfg(all(major_ge_3, minor_ge_5))]
+    pub fn set_no_data_value_i64(&mut self, no_data: Option<i64>) -> Result<()> {
+        let rv = if let Some(no_data) = no_data {
```
Comment (lnicola) [inline src/raster/rasterband.rs:799]: Well, isn't [this](https://github.com/georust/gdal/pull/308#discussion_r964887637) awkward...

Taking an `Option` seemed like a good idea at the time...

File: src/raster/tests.rs [hunk 5]
```diff
@@ -457,16 +458,57 @@ fn test_get_rasterband() {
...
+        let ds = driver.create_with_band_type::<i64, _>(&path, 1, 1, 1)?;
+        let mut rasterband = ds.rasterband(1)?;
+        assert_eq!(rasterband.no_data_value_i64(), None);
+        rasterband.set_no_data_value_i64(Some(255i64))?;
+    }
+
+    let ds = Dataset::open(&path)?;
+    let rasterband = ds.rasterband(1)?;
+
+    assert_eq!(rasterband.no_data_value_i64(), Some(255i64));
```
Comment (lnicola) [inline src/raster/tests.rs:489]: Should we use something larger here? I don't care that much.

File: src/raster/rasterband.rs (hunk 4 above)
Comment (metasim) [reply src/raster/rasterband.rs:799]: I still think there's value in keeping them consistent...

The alternative is to add [GDALDeleteRasterNoDataValue](https://gdal.org/api/raster_c_api.html#_CPPv427GDALDeleteRasterNoDataValue15GDALRasterBandH)

Comment (lnicola) [reply src/raster/rasterband.rs:799]: And now I think we should go for that (but in a separate PR).

File: src/raster/tests.rs (hunk 5 above)
Comment (metasim) [reply src/raster/tests.rs:489]: Sure, why not.

Are you ok with basically synthesizing the test data on the fly, rather than checking in files?

Comment (lnicola) [reply src/raster/tests.rs]: Using MEM? Sure, that's a good idea. We're not testing GDAL here,.we just want to make sure we don't crash.

Comment (metasim) [reply src/raster/tests.rs]: It's using `TempFixture`, which does actually write to disk, but the rational is the same. Just testing bindings.

**Review by lnicola — APPROVED** [review]

**Comment by lnicola** [comment]
Please squash at the end!

**Pull request merged**